vmdb/app/controllers/api_controller/vms.rb

```diff
@@ -64,6 +64,19 @@ def set_owner_resource_vms(type, id = nil, data = nil)
       end
     end
 
+    def add_event_resource_vms(type, id = nil, data = nil)
```

Comment:
target methods built dynamically and called if they exist. in this case (action_name)_resource_(collection_type).